UI for removing cart items (not including hooking up to API)

haszari opened the pull request:
Fixes #1534

This PR adds and styles UI elements for removing items from the cart. Note that these are not yet functional – we'll need to hook them up as part of #1489 . 

- On larger screens, there's a `Remove item` link under the quantity selector. This is already in master; this PR updates the styling.
- On smaller screens, there's a trashcan icon ([`delete_outline` from Material icons](https://material.io/resources/icons/?icon=delete_outline)), at the top right of the totals column.

#### Accessibility

- [x] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

Note: the color has been darkened to meet this contrast guideline, the design uses a lighter grey (cc @garymurray)

### Screenshots

<img width="851" alt="Screen Shot 2020-01-16 at 3 32 55 PM" src="https://user-images.githubusercontent.com/4167300/72488319-7a281e00-3875-11ea-967b-39e5750197a1.png">
<img width="450" alt="Screen Shot 2020-01-16 at 3 32 23 PM" src="https://user-images.githubusercontent.com/4167300/72488325-7e543b80-3875-11ea-8576-dd7fa519e70c.png">

Note: there's an issue with the quantity selector in these screenshots, logged here: #1578 

### How to test the changes in this Pull Request:

1. Clone this branch.
2. Add cart block to a page and publish. 
3. View the page on the front end and test in a variety of browsers/platforms/screen sizes.

nerrad commented on assets/js/blocks/cart-checkout/cart/full-cart/style.scss:
```diff
@@ -38,6 +38,21 @@ table.wc-block-cart-items {
+	line-height: 12px;
+	margin-top: 0.5em;
+	// Temporary - this is not yet a link or "link button".
+	// May not be needed when remove is hooked up to API properly.
```
I agree with @Aljullu  here (this is a [great resource](https://formidable.com/blog/2014/05/08/anchors-buttons-and-accessibility/) on this that I reference from time to time).

haszari replied:
Thanks for the nudge. I switched these over, which gives us a nice hover affordance for "free" (text/icon darkens on hover), and I put the resets in a mixin so we can easily reuse this approach. 61bd48deaf4c92eaf051758166840f8d6fb1a564

haszari replied:
<img width="441" alt="Screen Shot 2020-01-17 at 3 06 07 PM" src="https://user-images.githubusercontent.com/4167300/72578419-e7ea4d80-393a-11ea-863b-3a3dc88fb97f.png">

haszari replied:
(top one is darker because mouseover)

nerrad commented on assets/js/blocks/cart-checkout/cart/full-cart/cart-line-item-row.js:
```diff
@@ -132,14 +133,19 @@ const CartLineItemRow = ( { lineItem } ) => {
+				<div className="wc-block-cart-item__remove-icon">
+					<IconDelete />
+				</div>
+				<div className="wc-block-cart-item__totals-container">
```
Agree.

haszari replied:
Good point. This is a remnant from earlier iteration, when the totals were absolute positioned. ea41261f9deaa863e7487dcaeb47b1b4241e387f

nerrad commented on assets/js/components/icons/delete.js:
```diff
@@ -0,0 +1,24 @@
+// This is named `delete` as it is the `delete_outline` icon from Material.
+// https://material.io/resources/icons/?icon=delete_outline&style=baseline
+// We are using it as `trash` or `trashcan` or `remove-item`.
+export default () => (
```
Not sure on the name for this file.  I think we should be specific and call it `trash_outline` or something like that.

haszari replied:
Went with `IconTrash` - that's the simplest/shortest, and is a good combo of semantic (we use this for buttons which trash items) and visual (the icon is a trashcan). Didn't add `-outline` because that feels like an implementation detail (we only use outline version so far). 89d77ec3a1e088bf3064122e698a81a5b6a95bae

haszari commented:
Thanks for the reviews @Aljullu @nerrad :)

I've addressed all the feedback now, and also fixed some related issues with editor styles. Please take a look at these changes when re-reviewing. 71df8360283265824c3fcc4793372c93b79b8f73

Aljullu commented on assets/css/abstracts/_mixins.scss:
```diff
@@ -75,3 +75,18 @@
+	font-size: inherit;
+	font-weight: normal;
+
+	background: transparent;
```
This makes me think I would love to have a lint rule or a tool like CSS comb so we always order CSS properties in the same way. Is that opinion shared by other members of the team? Should we open an issue?

haszari replied:
I like this, I'm currently defaulting to "alphabetic unless I change my mind" :)

Aljullu commented on assets/css/abstracts/_mixins.scss:
```diff
@@ -75,3 +75,18 @@
 }
+
+// Reset <button> style so we can use link style for action buttons.
+@mixin link-button() {
```
Nice! This mixin will be handy in many ocasions. :clap: 

I wonder if it should be named `button-reset` instead. Technically it could be used in other occasions not only on link buttons.

haszari replied:
Works for me, though I'd probably tend to naming it based on a specific intention.

mikejolley commented:
Merging this so we can proceed with mobile styling/qty button fixes.
